tests/proc/lif/test_models.py

```diff
@@ -0,0 +1,350 @@
+    send_at_times: np.ndarray = LavaPyType(np.ndarray, bool, precision=1)
+
+    def run_spk(self):
+        """
+        Send `spikes_to_send` if current time-step requires it
+        """
+        if self.send_at_times[self.current_ts-1]:
+            self.s_out.send(self.vec_to_send)
+        else:
+            self.s_out.send(np.zeros_like(self.vec_to_send))
+
+
+@implements(proc=VecRecvProcess, protocol=LoihiProtocol)
+@requires(CPU)
+# need the following tag to discover the ProcessModel using LifRunConfig
+@tag('floating_pt')
+class PySpkRecvModel1(PyLoihiProcessModel):
+    s_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, bool, precision=1)
+    spk_data: np.ndarray = LavaPyType(np.ndarray, float)
+
+    def run_spk(self):
+        """Receive spikes and store in an internal variable"""
+        spk_in = self.s_in.recv()
+        self.spk_data[self.current_ts-1, :] = spk_in
+
+
+@implements(proc=VecRecvProcess, protocol=LoihiProtocol)
+@requires(CPU)
+# need the following tag to discover the ProcessModel using LifRunConfig
+@tag('fixed_pt')
+class PySpkRecvModel2(PyLoihiProcessModel):
+    s_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, bool, precision=1)
+    spk_data: np.ndarray = LavaPyType(np.ndarray, int, precision=1)
+
+    def run_spk(self):
+        """Receive spikes and store in an internal variable"""
+        spk_in = self.s_in.recv()
+        self.spk_data[self.current_ts-1, :] = spk_in
+
+
+class TestLIFProcessModels(unittest.TestCase):
+    """Tests for all ProcessModels of LIF"""
+    def test_float_pm_no_decay(self):
+        """
+        Set up bias = 1 * 2**1 = 2. and threshold = 4. Without any current
+        and voltage decay and in the absence of external input, bias driven
+        neurons spike every 2nd time-step.
+        """
+        shape = (10,)
+        num_steps = 10
+        sps = VecSendProcess(shape=shape, num_steps=num_steps,
+                             vec_to_send=np.zeros(shape, dtype=np.float),
+                             send_at_times=np.ones((num_steps,), dtype=np.bool))
+        lif = LIF(shape=shape,
+                  du=0.,
+                  dv=0.,
+                  bias=np.ones(shape, dtype=np.float),
+                  bias_exp=np.ones(shape, dtype=np.float),
+                  vth=4.)
+        spr = VecRecvProcess(shape=(num_steps, shape[0]))
+        sps.out_ports.s_out.connect(lif.in_ports.a_in)
+        lif.out_ports.s_out.connect(spr.in_ports.s_in)
+        rcnd = RunSteps(num_steps=num_steps)
+        rcfg = LifRunConfig(select_tag='floating_pt')
+        lif.run(condition=rcnd, run_cfg=rcfg)
+        spk_data_through_run = spr.spk_data.get()
+        lif.stop()
+        expected_spk_data = np.zeros((num_steps, shape[0]))
+        expected_spk_data[1:10:2, :] = 1.
+        self.assertTrue(np.all(expected_spk_data == spk_data_through_run))
+
+    def test_float_pm_impulse_du(self):
+        """
+        Send a single spike of magnitude 128 with a current decay of 0.5 and
```

Comment:
Docstring should say what it checks and maybe why this needs to be checked. 
Inline comments could also be used to explain the how along the way.

Reply:
Done.